src/CQRS/LeanCode.CQRS.Security/CustomAuthorizer.cs

```diff
@@ -1,29 +1,70 @@
+using System.Security.Claims;
 using Microsoft.AspNetCore.Http;
 
 namespace LeanCode.CQRS.Security;
 
-public interface ICustomAuthorizer
+public interface IHttpContextCustomAuthorizer
 {
-    Task<bool> CheckIfAuthorizedAsync(HttpContext httpContext, object obj, object? customData);
+    Task<bool> CheckIfAuthorizedAsync(HttpContext context, object obj, object? customData);
+}
+
+public interface ICustomAuthorizer : IHttpContextCustomAuthorizer
+{
+    Task<bool> CheckIfAuthorizedAsync(ClaimsPrincipal user, object obj, object? customData);
+
+    Task<bool> IHttpContextCustomAuthorizer.CheckIfAuthorizedAsync(
+        HttpContext context,
+        object obj,
+        object? customData
+    ) => CheckIfAuthorizedAsync(context.User, obj, customData);
+}
+
+public abstract class HttpContextCustomAuthorizer<TObject> : IHttpContextCustomAuthorizer
+{
+    public Task<bool> CheckIfAuthorizedAsync(HttpContext context, object obj, object? customData) =>
+        CheckIfAuthorizedAsync(context, (TObject)obj);
+
+    protected abstract Task<bool> CheckIfAuthorizedAsync(HttpContext context, TObject obj);
 }
 
 public abstract class CustomAuthorizer<TObject> : ICustomAuthorizer
 {
-    public Task<bool> CheckIfAuthorizedAsync(HttpContext httpContext, object obj, object? customData) =>
-        CheckIfAuthorizedAsync(httpContext, (TObject)obj);
+    public Task<bool> CheckIfAuthorizedAsync(ClaimsPrincipal user, object obj, object? customData) =>
+        CheckIfAuthorizedAsync(user, (TObject)obj);
+
+    protected abstract Task<bool> CheckIfAuthorizedAsync(ClaimsPrincipal user, TObject obj);
+}
+
+public abstract class HttpContextCustomAuthorizer<TObject, TCustomData> : IHttpContextCustomAuthorizer
```

Comment:
Overall, looks good, but can we get a set of (somewhat dummy) unit tests that check if the data is cast properly (and extracted properly)?

Reply:
Possibly, yes.